docker-setup/action.yml

```diff
@@ -0,0 +1,120 @@
+name: "Build and push docker"
+description: "Build and push docker image"
+
+inputs:
+  DOCKERHUB_USERNAME:
+    description: "Dockerhub username used to read public images"
+    required: true
+  DOCKERHUB_TOKEN:
+    description: "Dockerhub token used to read public images"
+    required: true
+
+  GITHUB_TOKEN:
+    description: "Token from secrets.GITHUB_TOKEN"
+    required: true
+
+  AWS:
+    description: "If we should use AWS"
+    default: true
+
+  AWS_ACCESS_KEY_ID:
+    description: "AWS access key used to write to ECR"
+    required: false
+  AWS_SECRET_ACCESS_KEY:
+    description: "AWS secret access key used to write to ECR"
+    required: false
+  AWS_REGION:
+    description: "AWS region"
+    required: false
+  AWS_ROLE_TO_ASSUME:
+    description: "AWS config to use"
+    required: false
+
+  ecr_registry:
+    default: ''
+    description: "ECR registry prefix"
+  ecr_type:
+    default: 'ecr'
+    description: "ecr or ecr-public"
+
+runs:
+  using: "composite"
+  steps:
+    - name: Login to Docker Hub
+      if: ${{ !github.event.pull_request.head.repo.fork }}
+      uses: docker/login-action@v2
+      with:
+        username: ${{ inputs.DOCKERHUB_USERNAME }}
+        password: ${{ inputs.DOCKERHUB_TOKEN }}
+
+    - name: Configure AWS
```

Comment:
Use official AWS actions?

https://github.com/aws-actions/configure-aws-credentials

Reply:
History is reason for this. Lets try if we can use it here.